modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/UriInfoUtil.java

```diff
@@ -42,9 +42,21 @@ public static UriBuilder getBaseUriBuilder(
 		UriBuilder uriBuilder = getBaseUriBuilder(uriInfo);
 
 		uriBuilder.host(PortalUtil.getForwardedHost(httpServletRequest));
-		uriBuilder.port(PortalUtil.getForwardedPort(httpServletRequest));
 
-		if (PortalUtil.isSecure(httpServletRequest)) {
+		int port = PortalUtil.getForwardedPort(httpServletRequest);
+
+		boolean secure = PortalUtil.isSecure(httpServletRequest);
+
+		if ((!secure && (port != Http.HTTP_PORT)) ||
+			(secure && (port != Http.HTTPS_PORT))) {
+
+			uriBuilder.port(port);
+		}
+		else {
+			uriBuilder.port(-1);
```

Comment:
Is this needed?

Reply:
yes because the uribuilder already has the port set at that point.